libs/cardano-ledger-core/src/Cardano/Ledger/BaseTypes.hs

```diff
@@ -344,6 +346,20 @@ fromRatioBoundedRatio ratio
     lowerBound = minBound :: BoundedRatio b a
     upperBound = maxBound :: BoundedRatio b a
 
+instance (ToCBOR a, Integral a, Bounded a, Typeable b) => ToCBOR (BoundedRatio b a) where
+  toCBOR (BoundedRatio u) = Plain.encodeRatioWithTag toCBOR u
+
+instance
+  (FromCBOR a, Bounded (BoundedRatio b a), Bounded a, Integral a, Typeable b, Show a) =>
+  FromCBOR (BoundedRatio b a)
+  where
+  fromCBOR = do
```

Comment:
This should be equivalent to `EncCBOR (BoundedRatio)`, in so far as  `decodeCollectionWithLen` [from cardano-binary](https://github.com/IntersectMBO/cardano-base/blob/56ed3e31560f5c9c34c2f602ccb278493a0d49ae/cardano-binary/src/Cardano/Binary/FromCBOR.hs#L584) is equivalent to `decodeCollectionWithLen` from [Decoder](https://github.com/IntersectMBO/cardano-ledger/blob/edb2ee48409871eae0687692994c11d38ffd261a/libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Decoder.hs#L736).
I think they are, but I hesitate to derive `DecCBOR` from this version.